File: app/src/main/java/com/example/tradeup/data/repository/UserSavedItemsRepository.java
package com.example.tradeup.data.repository;

import androidx.annotation.NonNull;
import com.example.tradeup.core.utils.Callback;
// import com.example.tradeup.data.model.Item; // Bỏ qua nếu chỉ làm việc với ID
import java.util.List;

public interface UserSavedItemsRepository {
    void saveItem(@NonNull String userId, @NonNull String itemId, Callback<Void> callback);

    void unsaveItem(@NonNull String userId, @NonNull String itemId, Callback<Void> callback);

    void getSavedItemIds(@NonNull String userId, Callback<List<String>> callback);

    void isItemSaved(@NonNull String userId, @NonNull String itemId, Callback<Boolean> callback);
}

File: app/src/main/java/com/example/tradeup/data/repository/UserSavedItemsRepository.kt
package com.example.tradeup.data.repository

import com.example.tradeup.data.model.Item // Có thể bạn muốn trả về danh sách Item thay vì chỉ ID

interface UserSavedItemsRepository {
    suspend fun saveItem(userId: String, itemId: String): Result<Unit>
    suspend fun unsaveItem(userId: String, itemId: String): Result<Unit>
    suspend fun getSavedItemIds(userId: String): Result<List<String>> // Hoặc Result<List<Item>> nếu muốn lấy cả object Item
    suspend fun isItemSaved(userId: String, itemId: String): Result<Boolean>
}